Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/ObjectCollectionNode.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    /// <summary>
+    /// Collection node is a data storage class for individual data about an object in a collection.
+    /// </summary>
+    [Serializable]
+    public class ObjectCollectionNode
```

Comment:
pretty sure we can probably make this a struct. Where are we checking if it's null?

Reply:
Its not the null check (it was unnecessary so I removed it), ObjectCollectionNode needs to be mutable. Switching to a struct broke basically every line where properties were being modified.

Reply:
All you'd need to do is get a local copy, assign the values, then assign your modified copy back to the array. Like so:

```
case ObjectOrientationSurfaceTypeEnum.Scatter:
// Get randomized planar mapping
// Calculate radius of each node while we're here
// Then use the packer function to shift them into place
for (int i = 0; i < NodeList.Count; i++)
{
    var node = NodeList[i];
    newPos = VectorExtensions.ScatterMapping(nodeGrid[i], Radius);
    var nodeCollider = node.transform.GetComponentInChildren<Collider>();

    if (nodeCollider != null)
    {
        // Make the radius the largest of the object's dimensions to avoid overlap
        Bounds bounds = nodeCollider.bounds;
        node.Radius = Mathf.Max(Mathf.Max(bounds.size.x, bounds.size.y), bounds.size.z) * 0.5f;
    }
    else
    {
        // Make the radius a default value
        // TODO move this into a public field ?
        node.Radius = 1f;
    }

    node.transform.localPosition = newPos;
    UpdateNodeFacing(node);
    NodeList[i] = node;
}
```

Reply:
Ah, thanks @StephenHodgson. 😄 Just curious, is there a perf reason for switching to a struct?